source/physics_lists/PetaloPhysics.cc

```diff
@@ -93,4 +102,28 @@ void PetaloPhysics::ConstructProcess()
     pmanager->RemoveProcess(cs);
   }
 
+
+  if (nest_) {
+    PetaloDetector* petalo = new PetaloDetector();
+    G4double e_field = petalo->FitEF(0, 0, 0);
+    PetaloPersistencyManager* pm =
+      dynamic_cast<PetaloPersistencyManager*>(G4VPersistencyManager::GetPersistencyManager());
+    pm->SetElectricField(e_field);
+    NEST::NESTcalc* petaloCalc = new NEST::NESTcalc(petalo);
+    NEST::NESTProc* theNESTScintillationProcess =
+      new NEST::NESTProc("S1", fElectromagnetic, petaloCalc, petalo);
+    theNESTScintillationProcess->SetDetailedSecondaries(true); // this is to use the full scintillation spectrum of LXe.
+    theNESTScintillationProcess->SetStackElectrons(true); //false if only light is collected
+
+    auto aParticleIterator = GetParticleIterator();
+    aParticleIterator->reset();
+    while ((*aParticleIterator)()) {
```

Comment:
It is not clear to me what are we doing in this loop. Why do we need to iterate over these particles?

Reply:
This iterator is going through all the particles that are defined and when it founds a particle to which the NEST process is applicable, it adds this process to the particle. This way, during the simulation G4 knows that this particle can undergo this process.

Reply:
I see, thank you!